slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/JdbcMiscTest.scala

```diff
@@ -63,14 +63,34 @@ class JdbcMiscTest extends AsyncTest[JdbcTestDB] {
       GetStatementParameters.map { csp => csp shouldBe sp }
 
     Action.seq(
-      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto)),
+      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null)),
       Action.seq(
-        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto)),
-        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.HoldCursorsOverCommit)).
+        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null)),
+        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.HoldCursorsOverCommit, null)).
           withStatementParameters(rsHoldability = ResultSetHoldability.HoldCursorsOverCommit),
-        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto))
+        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null))
       ).withStatementParameters(rsType = ResultSetType.ScrollInsensitive),
-      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto))
+      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null))
+    )
+  }
+
+  def testOverrideStatements = {
+    class T(tag: Tag) extends Table[Int](tag, u"t") {
+      def id = column[Int]("a")
+      def * = id
+    }
+    val t = TableQuery[T]
+
+    val a1 = t.filter(_.id === 1)
+    val a2 = t.filter(_.id === 2)
+
+    seq(
+      t.schema.create,
+      t ++= Seq(1, 2, 3),
+      a1.result.map(_ shouldBe Seq(1)),
+      a1.result.overrideStatements(a2.result.statements).map(_ shouldBe Seq(2)),
```

Comment:
It's the same for `overrideStatements` as for `statements`. Some Actions have multiple statements (or none) and special-casing these would complicate the different Action types in the profiles even further, especially since it may depend on the actual implementation of a driver whether it can perform an operation with a single statement. Having an Iterable everywhere was a compromise to keep it manageable.